src/epic.js

```diff
@@ -0,0 +1,83 @@
+import Q from 'q';
+import {
+  sendRequest,
+  validateDate,
+} from './util';
+
+export default function epic(): object {
+  function validateType(type: string): string {
+    const validTypes = ['natural', 'enhanced'];
+    const i = validTypes.indexOf(type.toLowerCase());
+    return i === -1 ? '' : validTypes[i];
+  }
+
+  function badRequest(message: string, deferred: object): undefined {
```

Comment:
this is good and should be used in all of the modules, so you should move it into `util.js`. I don't think `badRequest` is the best name though, as it's catching an error before a request is ever made. Perhaps `handleError` is more descriptive?